### DISCUSSION TITLE
Comparing two point clouds

### DISCUSSION THREAD

**Discussion by MartinPedersenpp** [isl-org/Open3D#4984] [Q&A]
Hello all

I am trying to do comparison between a reference point cloud and and point cloud object captured with a stereo camera.

Currently I am using the point cloud difference method to find how many points are left in the reference point cloud once I remove all points from the fresh point cloud that is within 0.0025 meters.
But I am struggling with objects that are similar in height like a 0.25L soda can and a 0.25L juice box. They are allmost the same height, but one is rectangular and the other is round. When I subtract the can from the juice box it tends to leave about the same amount of points, if not less than the actual juice box (due to noise from the camera).

Is there a way to also look at the shape of the point cloud or another way to compare point clouds?

**Comment by yuecideng** [comment]
Maybe you can try to fit clinder primitives from these two point clouds and compare the fitness/inliers of each. Plane fitting can be also considered.

**Reply by MartinPedersenpp** [reply]
Can you possibly elaborate i little? not code, but workflow would be fine.

**Reply by yuecideng** [reply]
You can use algorithm like `ransac_fit_plane` or `ransac_fit_cylinder` to process your two point clouds data. The results would be the primitives model parameters and the inliers index list, then you can compare the value of `len(inliers) / len(points)` to clarify the type of point clouds. The `rectangular` one would have larger ratio if using `ransac_fit_plane`, or lower ratio if using `ransac_fit_cylinder` .

**Reply by MartinPedersenpp** [reply]
Thank you
I will surely check it out and see what kind of results I get
EDIT: Do I need to use a third party library for this? when I google I can only find "pyransac"
